internal/invocation/stripe_executor_test.go

```diff
@@ -0,0 +1,139 @@
+package invocation
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/goleak"
+)
+
+type orderChecker struct {
+	*testing.T
+	previousCallArg int
+}
+
+func (oc *orderChecker) call(arg int) {
+	assert.Equal(oc.T, oc.previousCallArg+1, arg, "order of the tasks is not preserved")
```

Comment:
How about panicing here instead of `assert.Equal`? `T` is supposed to be accessed from the goroutine it was created.

Reply:
Oh didn't know that. Thanks, will do as you suggest